app/controllers/admin/calendar_areas_controller.rb

```diff
@@ -0,0 +1,39 @@
+module Admin
+  class CalendarAreasController < AdminController
+    load_and_authorize_resource
+
+    def index
+      @calendar_areas = @calendar_areas.where.not(parent_area_id: nil).order(:title)
+    end
+
+    def new
+    end
+
+    def create
+      terms = EnergySparks::CsvLoader.from_text(params.fetch(:terms) {{}}.fetch(:csv) {''})
+      CalendarAreaFactory.create(@calendar_area, terms)
+      if @calendar_area.persisted?
+        redirect_to admin_calendar_areas_path, notice: 'Calendar area created'
+      else
+        render :new
+      end
+    end
+
+    def edit
+    end
+
+    def update
+      if @calendar_area.update(calendar_area_params)
+        redirect_to admin_calendar_areas_path, notice: 'Calendar area updated'
+      else
+        render :edit
```

Comment:
it's always the edits which are lonely and untested ;)

Reply:
I'll add a little tweak to the test to make sure it's not all red and embarassed